alws/release_planner.py

```diff
@@ -0,0 +1,704 @@
+import asyncio
+import re
+import copy
+import typing
+from collections import defaultdict
+
+from sqlalchemy.future import select
+from sqlalchemy.orm import Session, selectinload
+
+from alws import models
+from alws.config import settings
+from alws.constants import ReleaseStatus, RepoType, PackageNevra
+from alws.crud import sign_task
+from alws.errors import (
+    DataNotFoundError,
+    EmptyReleasePlan,
+    MissingRepository,
+    ReleaseLogicError,
+    SignError,
+)
+from alws.schemas import release_schema
+from alws.utils.beholder_client import BeholderClient
+from alws.utils.debuginfo import is_debuginfo_rpm
+from alws.utils.modularity import IndexWrapper
+from alws.utils.parsing import get_clean_distr_name
+from alws.utils.pulp_client import PulpClient
+
+
+class ReleasePlanner:
+    def __init__(self, db: Session):
+        self._db = db
+        self.pkgs_mapping = None
+        self.repo_data_by_href = None
+        self.pkgs_nevra = None
+        self.debug_pkgs_nevra = None
+        self.existing_packages = None
+        self.latest_repo_versions = None
+        self.base_platform = None
+        self._beholder_client = BeholderClient(settings.beholder_host)
+        self._pulp_client = PulpClient(
+            settings.pulp_host,
+            settings.pulp_user,
+            settings.pulp_password
+        )
+
+    async def get_pulp_packages(
+        self,
+        build_ids: typing.List[int],
+        build_tasks: typing.List[int] = None,
+    ) -> typing.Tuple[typing.List[dict], typing.List[str], typing.List[dict]]:
+        src_rpm_names = []
+        packages_fields = ['name', 'epoch', 'version', 'release', 'arch']
+        pulp_packages = []
+
+        builds_q = select(models.Build).where(
+            models.Build.id.in_(build_ids)).options(
+            selectinload(
+                models.Build.source_rpms).selectinload(
+                models.SourceRpm.artifact),
+            selectinload(
+                models.Build.binary_rpms).selectinload(
+                models.BinaryRpm.artifact),
+            selectinload(models.Build.tasks).selectinload(
+                models.BuildTask.rpm_module
+            ),
+            selectinload(models.Build.repos)
+        )
+        build_result = await self._db.execute(builds_q)
+        modules_to_release = {}
+        for build in build_result.scalars().all():
+            for rpm_data in (build.source_rpms, build.binary_rpms):
+                for rpm in rpm_data:
+                    # Failsafe to not process logs
+                    if rpm.artifact.type != 'rpm':
```

Comment:
AFAIK neither binary_rpm nor source_rpm can contain logs

Reply:
It's old logic from releases, i'll check it